example.c

```diff
@@ -15,24 +15,67 @@
  * Lesser General Public License for more details.
  */
 
+/*
+ * Example of commands:
+ * libtinyiiod version:
+ * VERSION
+ *
+ * print xml:
+ * PRINT
+ *
+ * read channel attribute:
+ * READ adc INPUT voltage0 scale
+ *
+ * write channel attribute:
+ * WRITE adc OUTPUT voltage0 scale 3
+ * 66
+ */
+
 #include "tinyiiod.h"
 #include <errno.h>
 #include <signal.h>
 #include <stdio.h>
 #include <string.h>
 
-static void read_data(char *buf, size_t len)
+static long read_value(const char *str)
 {
+	char *end;
+	int32_t value = strtol(str, &end, 0);
+
+	if (end == str)
+		return -EINVAL;
+	else
+		return value;
+}
+
+static int32_t read_data(int32_t *instance_id, char *buf, size_t len)
+{
+	*instance_id = 0;
 	fread(buf, 1, len, stdin);
+	return 1;
```

Comment:
Usually the return is separated from the rest of the code with a blank line. Also return 1 in this case is considered a succes, are there any SUCCESS, FAIL defines somewhere in the network part ?